### PR TITLE
`create project` can now open in Positron

### LINKED ISSUE #10268: Running `quarto create project blog .` in Positron Terminal opens VSCode window
### Bug description

I'm making a new blog via the Quarto CLI inside the terminal in Positron. It looks like there is some auto-open-up-in-new-window capabilities, but I can't seem to have the new window be Positron--it always opens up VSCode.

### Steps to reproduce

1. Open up Positron
2. In Positron terminal, run  `quarto create project blog .` 
3. See VSCode being opened

### Expected behavior

I expected a new window of Positron to be opened (or no window). 

### Actual behavior

Opened up a new window of VSCode. 

### Your environment

- Positron version: 2024.06.1 (Universal) build 17
- Quarto version: 1.5.47
- MacOS M1

### Quarto check output

```
Quarto 1.5.47
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.47
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2022.10
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/isabelzimmerman/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.7
      Path: /Users/isabelzimmerman/.pyenv/versions/3.10.7/envs/positron/bin/python
      Jupyter: 5.3.0
      Kernels: pydemo, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.1
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library
      knitr: 1.47
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK
```

### PR DESCRIPTION
**Pull request by cderv**
This PR teaches Quarto about Positron as an editor for `quarto create project`

- Either to select using `quarto create project --open positron`
- Either to detect when command is ran into Positron directly

Help page shows the new support

````
❯ quarto create --help

Usage:   quarto create [type] [commands...]
Version: 99.9.9

Description:

  Create a Quarto project or extension

Options:

  -h, --help              - Show this help.
  --open        [editor]  - Open new artifact in this editor (positron, vscode, rstudio)
  --no-open               - Do not open in an editor

(...)
````

I checked the Windows paths for default installation of both installers. 

This solves #10268

### CODE REVIEW

File: src/command/create/editor.ts [hunk 1]
```diff
@@ -141,6 +146,66 @@ function vscodeEditorInfo(): EditorInfo {
...
+    );
+    editorInfo.actions.push(...pathActions);
+  } else if (Deno.build.os === "darwin") {
+    editorInfo.actions.push({
+      action: "which",
+      arg: "positron",
+    });
+
+    const pathActions = macosAppPaths(
+      "Positron.app/Contents/Resources/app/bin/positron",
```
Comment (cderv) [inline src/command/create/editor.ts:192]: @cscheid @gordonwoodhull I need confirmation I got the path correctly on Mac; 

You can also test the PR by running 
`quarto create project blog .` inside Positron Terminal. 

I tested on WIndows

File: src/core/platform.ts
Comment (cderv) [inline src/core/platform.ts:1]: I added `isPositronTerminal()` and `isPositron()` to match with other editor, as the env var is `POSITRON` and not terminal specific it seems. `TERM_PROGRAM` is `vscode` like for VSCODE so not helpful. 

I asked confirmation for the env var to check to Positron team.

File: src/command/create/editor.ts (hunk 1 above)
Comment (gordonwoodhull) [reply src/command/create/editor.ts:192]: I checked out the branch and tried `quarto create project blog .` inside the Positron terminal. (MacOS)

It still opened a new VScode window. 

Anything I can do to test further?

Comment (cderv) [reply src/command/create/editor.ts:192]: yes. Can you check the path I used for the application ? 

What does`which positron` returns on your side ?

Comment (cderv) [reply src/command/create/editor.ts]: Thanks @gordonwoodhull for the debugging !

**Comment by cderv** [comment]
Working on Windows (tested by me) and Mac (tested by @gordonwoodhull)

**Pull request merged**